apps/test/unit/javalab/PlaygroundTest.js

```diff
@@ -445,6 +447,44 @@ describe('Playground', () => {
     expect(onNewlineMessage).to.have.been.calledOnce;
   });
 
+  it('does not send click event if update is in progress', () => {
+    playground.handleSignal({value: PlaygroundSignalType.RUN});
+
+    const imageId = 'id';
+    playground.handleImageClick(imageId);
+    expect(onJavabuilderMessage).to.have.been.calledOnce;
+
+    onJavabuilderMessage.reset();
+    playground.handleImageClick(imageId);
+    // Should not be called again
+    expect(onJavabuilderMessage).to.have.not.been.called;
+
+    playground.handleSignal({value: PlaygroundSignalType.UPDATE_COMPLETE});
+    playground.handleImageClick(imageId);
+    // Should be able to send message after receiving UPDATE_COMPLETE
+    expect(onJavabuilderMessage).to.have.been.calledOnce;
+  });
+
+  it('re-enables click events after timeout', done => {
+    playground.handleSignal({value: PlaygroundSignalType.RUN});
+
+    const imageId = 'id';
+    playground.handleImageClick(imageId);
+    expect(onJavabuilderMessage).to.have.been.calledOnce;
+
+    onJavabuilderMessage.reset();
+    playground.handleImageClick(imageId);
+    // Should not be called again
+    expect(onJavabuilderMessage).to.have.not.been.called;
+
+    setTimeout(() => {
+      playground.handleImageClick(imageId);
+      // Should be able to send message after timeout
+      expect(onJavabuilderMessage).to.have.been.calledOnce;
+      done();
+    }, REENABLE_CLICK_EVENTS_TIMEOUT_MS + 1);
```

Comment:
Rather than waiting 1.5s on the test, use the Sinon timer mock: https://sinonjs.org/releases/latest/fake-timers/

Reply:
I'd also recommend using this for previous test in this file to prevent flakiness in case of a hung test.

Reply:
Oh nice! will change